### PR TITLE
fix: ACNA-1480 - support for concurrentActivations limit key

### PR DESCRIPTION
**Pull request by florind12**
## Description
The manifest spec limits keys we are supporting are:  

- timeout
- memorySize
- logSize
- concurrentActivations

This PR maps, concurrentActivations key to concurrency key. 

## Related Issue
https://github.com/adobe/aio-cli-plugin-app/issues/507

## How Has This Been Tested?
unit tests
replication steps from the mentioned issue. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by sandeep-paliwal — APPROVED** [review]
LGTM

**Comment by Himavanth** [comment]
Hi @florind12 ,
There seems to be some ambiguity here about `concurrentActivations` being the same as `concurrency` .

The docs at https://github.com/apache/openwhisk-wskdeploy/blob/master/specification/html/spec_actions.md#valid-limit-keys say `concurrentActivations` is supported and it is `per-namespace`.

The parsers in wskdeploy show `concurrentActivations` under `unsupported` category. https://github.com/apache/openwhisk-wskdeploy/blob/master/parsers/yamlparser.go#L64-L65
https://github.com/apache/openwhisk-wskdeploy/blob/master/parsers/manifest_parser.go#L810

Do you know if `concurrentActivations` is working with wskdeploy ?

**Comment by florind12** [comment]
@Himavanth 
Good observation. 
I suppose this is fine since the code changes gives room to use both keys:  `concurrentActivations` || `concurrency`
I think what matters is that Runtime API supports only the `concurrentActivations` key, this is the purpose of these code changes.
image
Adobe Runtime API: https://adobeioruntime.net/api/v1/docs/index.html?url=/api/v1/api-docs

**Comment by florind12** [comment]
@shazron @Himavanth 
New info: 
wskdeploy will parse the `concurrentActivations` key, but will ignore it:

`Warning: Action Limit [concurrentActivations] is currently not changeable. Ignoring...`

File: src/utils.js [hunk 1]
```diff
@@ -1006,15 +1006,13 @@ function createActionObject (fullName, manifestAction) {
...
 
   if (manifestAction.limits) {
-    const limits = {
+    const concurrencyLimit = manifestAction.limits.concurrentActivations || manifestAction.limits.concurrency
+    objAction.limits = {
       memory: manifestAction.limits.memorySize || 256,
       logs: manifestAction.limits.logSize || 10,
-      timeout: manifestAction.limits.timeout || 60000
+      timeout: manifestAction.limits.timeout || 60000,
+      ...concurrencyLimit && { concurrency: concurrencyLimit }
```
Comment (shazron) [inline src/utils.js:1014]: can you express this in more approachable logic, for maintainability purposes

**Review by shazron — CHANGES_REQUESTED** [review]

**Review by shazron — APPROVED** [review]

**Pull request merged**